packages/loopring_v3.1/contracts/impl/managers/DepositModuleManager.sol

```diff
@@ -0,0 +1,38 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "../modules/DepositModule.sol";
+
+import "../../iface/IExchangeModuleFactory.sol";
+import "./../CircuitManager.sol";
+
+/// @title DepositModuleManager
+/// @author Brecht Devos - <brecht@loopring.org>
+contract DepositModuleManager is IExchangeModuleFactory, CircuitManager
+{
+    function createModule(
+        address exchangeAddress
+        )
+        external
+        returns (address)
+    {
+        // Can deploy the module using a proxy (if supported), cloning,...
```

Comment:
An exchange could write a wrapper contract that forwards all function calls to the correct exchange module (for example, it could still use the `blockType` function parameter in commitBlocks but there would be a lookup to the module instance, etc...). This could use a similar design pattern as the Relayer in the conditional transfer PR to do this with minimal effort with a proxy like construction, or every function can be manually wrapped and forwarded (as module changes will probably be infrequent).